src/utils/settings/splitFilters.js

```diff
@@ -0,0 +1,111 @@
+import { STANDALONE_MODE } from '../constants';
+import { uniq, isString } from '../lang';
+import logFactory from '../logger';
+const log = logFactory('splitio-settings');
+
+/**
+ * Validate, deduplicate and sort a given list of filter values.
+ * Exposed for testing purposes.
+ *
+ * @param {string} filterName string used for log messages
+ * @param {string[]} list list to validate, deduplicate and sort
+ * @param {number} maxLength
+ * @returns undefined or a sanitized list of values. The list is a non-empty array of unique and alphabetically sorted non-empty strings.
+ * @throws Error if the sanitized list exceeds the length indicated by `maxLength`
+ */
+function filterValidation(filterName, list, maxLength) {
+  // remove invalid values (no strings and empty strings)
+  let result = list.filter(value => {
+    if (!isString(value) || value === '') {
+      log.warn(`Malformed value in '${filterName}' filter ignored: '${value}'`);
+      return false;
+    }
+    return true;
+  });
+
+  // remove duplicated values
+  result = uniq(result);
+
+  // sort values
+  result.sort();
+
+  // check length
+  if (result.length > maxLength) throw new Error(`${maxLength} unique values can be specified at most for '${filterName}' filter. You passed ${result.length}. Please consider reducing the amount or using other filter.`);
+  if (result.length === 0) {
+    log.warn(`Ignoring ${filterName} filter. It has no valid values (no-empty strings).`);
+    return;
+  }
+  return result;
+}
+
+/**
+ * Returns a string representing the URL encoded query component of /splitChanges URL.
+ * Exposed for testing purposes.
+ * The possible formats of the query string are:
+ *  - '': if both byNameList and byPrefixList are undefined
+ *  - 'names=<comma-separated-values>': if only byPrefixList is undefined
+ *  - 'prefixes=<comma-separated-values>': if only byNameList is undefined
+ *  - 'names=<comma-separated-values>&prefixes=<comma-separated-values>': if no one is undefined
+ *
+ * @param {string[] | undefined} byNameList undefined or list of ordered and unique non-empty strings
+ * @param {string[] | undefined} byPrefixList undefined or list of ordered and unique non-empty strings
+ * @returns undefined or string with the `query` component of URL.
+ */
+function queryStringBuilder(byNameList, byPrefixList) {
```

Comment:
Method should receive the whole filters set/collection/array/wthv and know how to manage them, to avoid adding an extra param per filter type in the future.